app/scripts/ui.js

```diff
@@ -84,95 +80,35 @@ async function start() {
 
   // identify window type (popup, notification)
   const windowType = getEnvironmentType();
+  const extensionPort = browser.runtime.connect({ name: windowType });
 
-  // setup stream to background
-  extensionPort = browser.runtime.connect({ name: windowType });
-
-  let connectionStream = new PortStream(extensionPort);
-
-  const activeTab = await queryCurrentActiveTab(windowType);
-
-  /*
-   * In case of MV3 the issue of blank screen was very frequent, it is caused by UI initialising before background is ready to send state.
-   * Code below ensures that UI is rendered only after "CONNECTION_READY" or "startUISync"
-   * messages are received thus the background is ready, and ensures that streams and
-   * phishing warning page load only after the "startUISync" message is received.
-   * In case the UI is already rendered, only update the streams.
-   */
-  const messageListener = async (message) => {
-    const method = message?.data?.method;
-
-    switch (method) {
-      case METHOD_START_UI_SYNC:
-        await handleStartUISync();
-        break;
-      case METHOD_DISPLAY_STATE_CORRUPTION_ERROR:
-        handleDisplayStateCorruptionError(message.data.params);
-        break;
-      default:
-    }
-  };
+  const connectionStream = new PortStream(extensionPort);
+  const subStreams = connectSubstreams(connectionStream);
+  const backgroundConnection = metaRPCClientFactory(subStreams.controller);
+  connectToBackground(container, backgroundConnection, handleStartUISync);
+  setupProviderConnection(subStreams.provider);
 
   async function handleStartUISync() {
     endTrace({ name: TraceName.BackgroundConnect });
 
-    if (isManifestV3 && isUIInitialised) {
-      // Currently when service worker is revived we create new streams
```

Comment:
This is all removed because service workers aren't "revived" anymore. Please correct me if I'm wrong!